packages/relay/src/ServerConfigParams.ts

```diff
@@ -369,6 +371,20 @@ export function validateBalanceParams (config: ServerConfigParams): void {
   }
 }
 
+export async function validateHub (hubFile: string, hubAddress: string, managerAddress: string, workerAddress: string, contractInteractor: ContractInteractor): Promise<void> {
+  const managerNonce = await contractInteractor.getTransactionCount(managerAddress, 'pending')
+  const workerNonce = await contractInteractor.getTransactionCount(workerAddress, 'pending')
```

Comment:
when verifying that worker is "pristine", verify its BALANCE is also zero: the end of registration always put some eth into a worker, and that (along with transactionCount) is a great marker that it should not be re-used in another relayhub

Reply:
I don't think so: 
calling addWorker() if the worker is not "pristine" is an error, regardless of how you got there. 
your code relies on the "relayhub" lock file. if the admin removes it, then its OK to create a penalizeable worker. 
I say that even if the admin remove the lock file, it is STILL forbidden to reuse this worker.

Reply:
I would agree if that caught all of these errors, but that's not the case, the worker might still be penalizeable. So I'm not sure about adding an extra check to cover only some errors, if the admin really wants to lose his money and remove internal files AND change the hub address in the config. That's really taking an extra step to sabotage yourself. 
@forshtat ?